### DISCUSSION TITLE
Why does the point/sun resolution affect my shadow when I only have spot light

### DISCUSSION THREAD

**Discussion by HSDSZ** [bnpr/Malt#380] [Q&A]
![image](https://user-images.githubusercontent.com/36077492/185194455-d0115883-ed33-4811-b301-137cc1527537.png)
![image](https://user-images.githubusercontent.com/36077492/185194521-83dcd8e4-ebf6-4330-b905-4109e6a3695a.png)

The shadow is still there when sun and point light resolutions are set to 1 but disappear when any one of them is 0. Is this what is supposed to be like or a bug?

If I only have spot light in my scene, sun resolution is set to 1. Will the render be faster if I further decrease the sun max distance and sun csm counts?

**Comment by pragma37** [comment]
Malt always creates at least one shadow map for each light type, so if you set it to 0 it actually tries a 0 resolution texture and OpenGL fails.
I've updated the dev branch so a minimum size is always ensured.

> If I only have spot light in my scene, sun resolution is set to 1. Will the render be faster if I further decrease the sun max distance and sun csm counts?

If you don't have any sun in your scene then nope.
Decreasing csm count would decrease vram consumption, but at 1x1 resolution we are talking about less than 1kB in total.
